alerter/src/androidTest/java/com/tapadoo/alerter/AlerterTest.kt

```diff
@@ -0,0 +1,131 @@
+package com.tapadoo.alerter
+
+import android.graphics.drawable.ColorDrawable
+import android.os.Build
+import android.support.test.filters.LargeTest
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import android.support.v4.content.ContextCompat
+import android.util.Log
+import android.view.View
+import android.view.ViewGroup
+
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Alerter Test Case Class
+ *
+ * @author Kevin Murphy, Tapadoo
+ * @since 05/10/2016
+ */
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class AlerterTest {
+
+    //Rule which sets the Activity to be used
+    @Rule
+    internal val activityRule = ActivityTestRule(MockActivity::class.java)
+
+    @Test
+    fun testConstruction() {
+        val alerter = Alerter.create(activityRule.activity)
+        Assert.assertNotNull(alerter)
+    }
+
+    @Test
+    fun testShow() {
+        //Get the Activity's initial number of Child Views
+        val childCount = (activityRule.activity.findViewById<View>(android.R.id.content).rootView as ViewGroup).childCount
+
+        //Instantiate Alerter
+        val alerter = Alerter.create(activityRule.activity)
+
+        //Show it to the User
+        alerter.show()
+
+        //Wait to ensure Alert layout is fully added
+        try {
+            Thread.sleep(1000)
+        } catch (e: InterruptedException) {
+            Log.e(javaClass.simpleName, Log.getStackTraceString(e))
+        }
+
+        //Get the new child count - Should be initial + 1
+        val childCountAfter = (activityRule.activity.findViewById<View>(android.R.id.content).rootView as ViewGroup).childCount
+
+        //Check to ensure we have an additional child added
+        Assert.assertNotSame(childCount, childCountAfter)
+        Assert.assertSame(childCount + 1, childCountAfter)
+    }
+
+    @Test
+    fun testBuilderStrings() {
+        val alert = Alerter.create(activityRule.activity)
+                .setTitle("Hello")
+                .setText("Hi")
+                .show()
+
+        Assert.assertSame(alert?.title?.text, "Hello")
```

Comment:
Should the alerter be null here after just being created? It might not be shown yet, but that shouldn't mean its null.